benchmark/src/main/java/ai/timefold/solver/benchmark/impl/report/LineChart.java

```diff
@@ -209,25 +223,35 @@ public LineChart<X, Y> build(String fileName, String title, String xLabel, Strin
              * This allows Chart.js to only draw the absolute minimum necessary points.
              */
             data.values().forEach(map -> {
-                List<Map.Entry<X, Y>> entries = map.entrySet().stream().toList();
+                List<Entry<X, Y>> entries = map.entrySet().stream().toList();
                 if (entries.size() < 3) {
                     return;
                 }
                 for (int i = 0; i < entries.size() - 2; i++) {
-                    Map.Entry<X, Y> entry1 = entries.get(i);
-                    Map.Entry<X, Y> entry2 = entries.get(i + 1);
+                    Entry<X, Y> entry1 = entries.get(i);
+                    Entry<X, Y> entry2 = entries.get(i + 1);
                     if (!entry1.getValue().equals(entry2.getValue())) {
                         continue;
                     }
-                    Map.Entry<X, Y> entry3 = entries.get(i + 2);
+                    Entry<X, Y> entry3 = entries.get(i + 2);
                     if (entry2.getValue().equals(entry3.getValue())) {
                         map.remove(entry2.getKey());
                     }
                 }
             });
-            // Then find all points on the X axis across all data sets.
+            /*
+             * Sometimes, when the dataset size is large, it can cause the browser to freeze or use excessive memory
+             * while rendering the line chart. To solve the issue of a large volume of data points, we use the
+             * Largest-Triangle-Three-Buckets algorithm to down-sample the data.
+             */
+            Map<String, Map<X, Y>> datasetMap = new LinkedHashMap<>(data.size());
+            for (Entry<String, NavigableMap<X, Y>> entry : data.entrySet()) {
```

Comment:
Another great place for `var`. Makes the code much more readable.
In fact, I'd go as far as saying that we should make `var` the default, and use the full type if we have a reason to do that.